python/pyspark/sql/session.py

```diff
@@ -305,10 +305,9 @@ def __init__(
             ):
                 jsparkSession = self._jvm.SparkSession.getDefaultSession().get()
             else:
-                jsparkSession = self._jvm.SparkSession(self._jsc.sc())
-                if options is not None:
-                    for key, value in options.items():
-                        jsparkSession.sharedState().conf().set(key, value)
+                jsparkSession = self._jvm.SparkSession(
+                    self._jsc.sc(), self._jvm.PythonUtils.toScalaMap(options)
```

Comment:
Looks good but last question. is this to ScalaMap required? We can make the Scala side to take Java map

Reply:
> Looks good but last question. is this to ScalaMap required? We can make the Scala side to take Java map

It seems worthwhile to use Java map to reduce one call of py4j.  Although Java map here looks not consistent.